### PR TITLE
Remove support for the Weather Underground API

### LINKED ISSUE #2391: Depricate the Weather Underground API
The Weather Underground API will be deactivated on 31/12/2018. As such support for the weather underground API should be de-activated. 

For the Weather Underground announcement see: https://apicommunity.wunderground.com/weatherapi/topics/end-of-service-for-the-weather-underground-api

### PR DESCRIPTION
**Pull request by tantalum**
Remove the Weather Underground API to resolve #2391

### CODE REVIEW

**Comment by dsander** [comment]
Thanks! We have to wait until the end of the year before we merge this, because the API stays [online until then](https://apicommunity.wunderground.com/weatherapi/topics/end-of-service-for-the-weather-underground-api).

**Comment by dsander** [comment]
Wunderground moved the shutdown date to the 15th of Feb, which was yesterday, but for some reason my WeatherAgent is still working ¯\_(ツ)_/¯

File: app/models/agents/weather_agent.rb [hunk 1]
```diff
@@ -135,98 +108,82 @@ def validate_location
     end
 
     def validate_options
-      errors.add(:base, "service must be set to 'darksky' or 'wunderground'") unless wunderground? || dark_sky?
```
Comment (dsander) [inline app/models/agents/weather_agent.rb]: I think it would be nice to show an error message that the wunderground API shut down if an Agent is still configured as `wunderground?`.

File: app/models/agents/weather_agent.rb [hunk 2]
```diff
@@ -135,98 +108,82 @@ def validate_location
...
-        forecast['forecast']['txt_forecast']['forecastday'].each { |daily| (merged[daily['period']] || {}).merge!(daily) }
-        merged
-      end
-    end
-
     def dark_sky
       if key_setup?
         ForecastIO.api_key = interpolated['api_key']
         lat, lng = coordinates
-        ForecastIO.forecast(lat, lng, params: {lang: language.downcase})['daily']['data']
```
Comment (dsander) [inline app/models/agents/weather_agent.rb]: Why did you remove the language argument?

**Comment by e-alfred** [comment]
The API doesn't work anymore after 5th of March. Please remove the code and maybe even add a better alternative like openweathermap.org.

File: app/models/agents/weather_agent.rb (hunk 2 above)
Comment (tantalum) [reply app/models/agents/weather_agent.rb]: I removed the language argument because it was causing the request to fail.

File: app/models/agents/weather_agent.rb (hunk 1 above)
Comment (tantalum) [reply app/models/agents/weather_agent.rb]: I added a check to display an error message if Weather Underground is still configured.

File: app/models/agents/weather_agent.rb (hunk 2 above)
Comment (dsander) [reply app/models/agents/weather_agent.rb]: Can you please check it again? Their API docs still list the `lang` parameter https://darksky.net/dev/docs#forecast-request

Comment (tantalum) [reply app/models/agents/weather_agent.rb]: Yeah I can check it again.

Comment (tantalum) [reply app/models/agents/weather_agent.rb]: I added in the `language` parameter and it did work.

File: app/models/agents/weather_agent.rb [hunk 3]
```diff
@@ -111,14 +90,17 @@ def coordinates
     end
 
     def language
-      interpolated['language'].presence || 'EN'
```
Comment (dsander) [inline app/models/agents/weather_agent.rb]: Looks like there are 4 instead of 2 spaces used for the indentation.

File: spec/models/agents/weather_agent_spec.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@
     Agents::WeatherAgent.create(
       name: 'weather',
       options: {
-        :location => 94103,
```
Comment (dsander) [inline spec/models/agents/weather_agent_spec.rb]: We use 2 spaces for indentation.

File: app/models/agents/weather_agent.rb (hunk 2 above)
Comment (dsander) [reply app/models/agents/weather_agent.rb]: That is great!

File: app/models/agents/weather_agent.rb (hunk 3 above)
Comment (tantalum) [reply app/models/agents/weather_agent.rb]: Will fix

Comment (tantalum) [reply app/models/agents/weather_agent.rb]: Fixed

File: spec/models/agents/weather_agent_spec.rb (hunk 4 above)
Comment (tantalum) [reply spec/models/agents/weather_agent_spec.rb]: Fixed

File: app/models/agents/weather_agent.rb [hunk 5]
```diff
@@ -135,22 +117,12 @@ def validate_location
     end
 
     def validate_options
-      errors.add(:base, "service must be set to 'darksky' or 'wunderground'") unless wunderground? || dark_sky?
+      errors.add(:base, "The Weather Underground API has been disables since Jan 1st 2018, please switch to DarkSky") if wunderground?
```
Comment (knu) [inline app/models/agents/weather_agent.rb]: There's a trivial typo `disables` and maybe the date should be updated.

Comment (tantalum) [reply app/models/agents/weather_agent.rb]: Good cach ;-). I just pushed an update

**Comment by virtadpt** [comment]
I think that should be "disabled", not "disables".

**Comment by knu** [comment]
Thanks!
